Source/msg.cpp

```diff
@@ -1785,7 +1785,7 @@ size_t OnPlayerDamage(const TCmd *pCmd, Player &player)
 
 	Player &target = Players[message.bPlr];
 	if (&target == MyPlayer && leveltype != DTYPE_TOWN && gbBufferMsgs != 1) {
-		if (player.isOnActiveLevel() && damage <= 192000 && target._pHitPoints >> 6 > 0) {
+		if (player.isOnActiveLevel() && damage <= 192000 && target._pHitPoints >> 6 > 0 && !target.friendlyMode) {

```

Comment:
I think if we don't check both players, we could be creating an opening for CheatEngine users.
```suggestion
		if (player.isOnActiveLevel() && damage <= 192000 && target._pHitPoints >> 6 > 0 && !player.friendlyMode && !target.friendlyMode) {
```